README.md

![TypoPRO](etc/logo.jpg "TypoPRO")

TypoPRO 2.5.0
=============

http://typopro.org/

*Fonts for Professional Typography &mdash; The Non-Cloud Way!*

**TypoPRO** is a carefully hand-selected collection of freely usable
professional fonts for use in Desktop Publishing, on Websites and
especially inside Web-based Desktop or Mobile Applications. Currently,
**TypoPRO** consists of 834 individual fonts of 114 font families. The
fonts in total span a very wide range of font types, font styles, font
weights and font variants. This way, **TypoPRO** provides a magnitude of
typographic possibilities with just a minimum number of font families.

Features
--------

- **Preselection** & **Assembling**:
  The fonts were carefully hand-selected, manually downloaded and
  consistently assembled into the **TypoPRO** collection. For many fonts
  the real upstream vendor origin of their latest versions had to be
  determined first, too.

- **Glyph Design**:
  All fonts are professionally crafted and have their glyphs (individual
  characters) beautifully designed and this way provide a really
  wonderful optical appearance. Most of the fonts are actually from
  professional font foundries.

- **Font Families**:
  Most of the fonts are based on reasonably large font families of
  individual fonts. This is important, because for professional
  typography results you usually always need multiple weights of at
  least the headline and text fonts.

- **Kerning**:
  Most of the fonts carry reasonable kerning information, provided by the
  font foundry: instructions for spacing adjustments between particular
  pairs of glyphs. This is important, because for professional
  typography, kerning is essential.

- **Hinting**:
  All fonts carry reasonable hinting information, provided by the
  conversion process (and its `ttfautohint` utility) of **TypoPRO**:
  display instructions for rasterization adjustments at various
  rendering sizes. This is important, because for rendering fonts on the
  Web at very small sizes, hinting is essential.

- **Liberal Licensing**:
  All fonts are under very liberal Open Source licenses and can be
  freely and unrestrictively used even in Closed Source and Commercial
  contexts.

- **Web Conversion**:
  All fonts are consistently converted from their original TTF/OTF
  format into various font formats for use on the Web: Embedded OpenType
  (EOT), Web Open Font Format (WOFF) and TrueType Font (TTF). These
  different formats for each font are required for maximum portability
  across browsers and operating systems. Suitable corresponding CSS
  `@font-face` declaration files were generated, too.

- **Unicode Character Reduction**
  Some fonts contain a really large number of glyphs, covering a wide
  range of the Unicode character set. This is great in general, but
  dramatically increases the font size, which is a major drawback on the
  web. For this reason, although it reduces the fonts to be used for
  the main Latin based languages (English, German, etc), we reduce the
  fonts in web formats to the smaller Unicode range 0000-00FF (Basic
  Latin, Latin-1 Supplement). For the DTP formats of all fonts we reduce
  to the larger (but still reduced for some very large fonts) Unicode
  range set 0000-036F,1DC0-1EFF,2000-20FF,2150-218F,2C60-2C7F,A720-A7FF
  (Basic Latin, Latin-1 Supplement, Latin Extended-A, Latin Extended-B,
  IPA Extensions, Spacing Modifier Letters, Combining Diacriticals
  Marks, Combining Diacritical Marks Supplement, Latin Extended
  Additional, General Punctuation. Superscripts and Subscripts, Currency
  Symbols, Combining Diacritical Marks for Symbols, Number Forms, Latin
  Extended-C, Latin Extended-D)

- **Font Renaming**:
  To circumvent license restrictions related to font format conversions
  and to avoid confusion with the original font formats, all Web and DTP font
  format variants were renamed to consistently carry the `TypoPRO `
  prefix.

- **Font Parameters**:
  In the Web font format variants, all font families consistently use
  a common font family name (CSS `font-family` attribute) and all
  fonts within the same font family are consistently distinguished
  via CSS parameters `font-weight`, `font-style`, `font-stretch` and
  `font-variant`.

- **Specimen**:
  For the Web font variants, a specimen in HTML format was generated to
  conveniently preview the font. See http://typopro.org/specimen/ for
  the current overview of all specimens in **TypoPRO**.

Font Overview
-------------

![The TypoPRO Font Cloud Image](etc/cloud.png "The TypoPRO Font Cloud Image")

#### General Purpose Fonts

The fonts in this group are general purpose ones, i.e., they can be used
equally well for headlines, body text, footnotes, etc. They especially
are part of whole font families and cover a wide range of Unicode
characters.

Name                      |Fonts|Serif|Slab|Sans|Mono|Hand|Regular|Italic|Cond|Thin|Medium|Normal|Bold|Black
--------------------------|----:|:---:|:--:|:--:|:--:|:--:|:-----:|:----:|:--:|:--:|:----:|:----:|:--:|:---:
Aileron                   | 16  |     |    | X  |    |    |  X    |  X   |    | X  |  X   |  X   | X  | X
Alegreya                  | 40  |  X  |    | X  |    |    |  X    |  X   |    | X  |      |  X   | X  |
Aleo                      |  6  |     | X  |    |    |    |  X    |  X   |    | X  |      |  X   | X  |
Amble                     |  8  |     |    | X  |    |    |  X    |  X   | X  | X  |      |  X   | X  |
Andika&nbsp;New&nbsp;Basic|  4  |  X  |    |    |    |    |  X    |  X   |    |    |      |  X   | X  |
Anonymous&nbsp;Pro        |  4  |     |    |    | X  |    |  X    |  X   |    |    |      |  X   | X  |
Asap                      |  4  |     |    | X  |    |    |  X    |  X   |    |    |      |  X   | X  |
Bitter                    |  4  |     | X  |    |    |    |  X    |  X   |    |    |      |  X   | X  |
Cabin                     | 14  |     |    | X  |    | X  |  X    |  X   | X  |    |  X   |  X   | X  |
Charis&nbsp;SIL           |  4  |  X  |    |    |    |    |  X    |  X   |    |    |      |  X   | X  |
Clear&nbsp;Sans           |  8  |     |    | X  |    |    |  X    |  X   |    | X  |  X   |  X   | X  |
Comme                     | 10  |     |    | X  | X  |    |  X    |  X   |    | X  |  X   |  X   | X  | X
Cooper&nbsp;Hewitt        | 14  |     |    | X  |    |    |  X    |  X   |    | X  |  X   |  X   | X  | X
Courier&nbsp;Prime        |  4  |     |    |    | X  |    |  X    |  X   |    |    |      |  X   | X  |
Cousine                   |  4  |     |    |    | X  |    |  X    |  X   |    |    |      |  X   | X  |
Crete&nbsp;Round          |  2  |     | X  |    |    |    |  X    |  X   |    |    |      |  X   |    |
Crimson                   |  6  |  X  |    |    |    |    |  X    |  X   |    |    |      |  X   | X  |
Cuprum                    |  4  |     |    | X  |    |    |  X    |  X   |    |    |      |  X   | X  |
DejaVu                    | 21  |  X  |    | X  | X  |    |  X    |  X   | X  | X  |      |  X   | X  |
Domine                    |  2  |  X  |    |    |    |    |  X    |      |    |    |      |  X   | X  |
Dosis                     |  7  |     |    | X  |    |    |  X    |  X   | X  | X  |  X   |  X   | X  | X
Droid                     |  7  |  X  |    | X  | X  |    |  X    |  X   |    |    |      |  X   | X  |
EB&nbsp;Garamond          |  4  |  X  |    |    |    |    |  X    |  X   |    |    |      |  X   | X  |
Ek&nbsp;Mukta             |  7  |     |    | X  |    |    |  X    |      |    | X  |  X   |  X   | X  | X
Encode&nbsp;Sans          | 45  |     |    | X  |    |    |  X    |      | X  | X  |  X   |  X   | X  | X
Fantasque&nbsp;Sans&nbsp;Mono|4 |  X  |    |    | X  |    |  X    |  X   |    |    |      |  X   | X  |
Fira&nbsp;Mono            |  2  |     |    |    | X  |    |  X    |      |    |    |      |  X   | X  |
Fira&nbsp;Sans            | 32  |     |    | X  |    |    |  X    |  X   |    | X  |  X   |  X   | X  | X
Gentium                   |  8  |  X  |    |    |    |    |  X    |  X   |    |    |      |  X   | X  |
Inconsolata               |  1  |     |    |    | X  |    |  X    |      |    |    |      |      | X  |
Latin&nbsp;Modern         | 50  |  X  |    | X  | X  |    |  X    |  X   | X  | X  |  X   |  X   | X  | X
Lato                      | 10  |     |    | X  |    |    |  X    |  X   |    | X  |  X   |  X   | X  | X
Lekton                    |  3  |     |    | X  | X  |    |  X    |  X   |    |    |      |  X   | X  |
Liberation                | 12  |  X  |    | X  | X  |    |  X    |  X   |    |    |      |  X   | X  |
Libre&nbsp;Baskerville    |  3  |  X  |    |    |    |    |  X    |  X   |    |    |      |  X   | X  |
Libre&nbsp;Caslon         |  4  |  X  |    |    |    |    |  X    |  X   |    |    |      |  X   | X  |
Lora                      |  4  |  X  |    |    |    |    |  X    |  X   |    |    |      |  X   | X  |
Merriweather              |  8  |  X  |    |    |    |    |  X    |  X   |    | X  |  X   |  X   | X  | X
Merriweather&nbsp;Sans    |  8  |     |    | X  |    |    |  X    |  X   |    | X  |  X   |  X   | X  | X
Montserrat                |  4  |  X  |    |    |    |    |  X    |      |    |    |      |  X   | X  |
Mplus                     | 43  |     |    | X  |    |    |  X    |      |    | X  |  X   |  X   | X  | X
Muli                      | 10  |     |    | X  |    |    |  X    |  X   |    | X  |  X   |  X   | X  | X
Neuton                    |  7  |  X  |    |    |    |    |  X    |  X   |    | X  |  X   |  X   | X  | X
Noto&nbsp;Sans            |  4  |     |    | X  |    |    |  X    |  X   |    |    |      |  X   | X  |
Noto&nbsp;Serif           |  4  |  X  |    |    |    |    |  X    |  X   |    |    |      |  X   | X  |
Open&nbsp;Sans            | 13  |     |    | X  |    |    |  X    |  X   | X  |    |  X   |  X   | X  | X
Oswald                    | 14  |     |    | X  |    |    |  X    |  X   |    | X  |  X   |  X   | X  | X
Overlock                  |  4  |     |    | X  |    |    |  X    |  X   |    |    |      |  X   | X  |
Oxygen                    |  5  |     |    | X  | X  |    |  X    |  X   |    |    |      |  X   | X  |
Poly                      |  2  |  X  |    |    |    |    |  X    |  X   |    |    |      |  X   |    |
Quattrocento              |  6  |  X  |    |    | X  |    |  X    |  X   |    |    |      |  X   | X  |
Quicksand                 |  6  |     |    | X  |    |    |  X    |  X   |    |    |      |  X   | X  |
Raleway                   | 18  |     |    | X  |    |    |  X    |  X   | X  | X  |  X   |  X   | X  | X
Roboto                    | 16  |     |    | X  | X  |    |  X    |  X   | X  | X  |  X   |  X   | X  |
Rosario                   |  4  |     |    | X  |    |    |  X    |  X   |    |    |      |  X   | X  |
Signika                   |  4  |     |    | X  |    |    |  X    |  X   |    |    |      |  X   | X  |
Sinkin&nbsp;Sans          | 18  |     |    | X  |    |    |  X    |  X   |    | X  |  X   |  X   | X  | X
Source&nbsp;Code&nbsp;Pro |  7  |     |    |    | X  |    |  X    |  X   |    | X  |  X   |  X   | X  | X
Source&nbsp;Sans&nbsp;Pro | 12  |     |    | X  |    |    |  X    |  X   |    | X  |  X   |  X   | X  | X
Source&nbsp;Serif&nbsp;Pro|  3  |  X  |    |    |    |    |  X    |      |    | X  |  X   |  X   | X  | X
TeX&nbsp;Gyre             | 33  |  X  |    | X  | X  | X  |  X    |  X   | X  | X  |  X   |  X   | X  |
Titillium                 | 11  |     |    | X  |    |    |  X    |  X   | X  | X  |  X   |  X   | X  | X
Ubuntu                    | 13  |     |    | X  | X  |    |  X    |  X   | X  | X  |  X   |  X   | X  |
Vegur                     |  3  |     |    | X  |    |    |  X    |      |    |    |  X   |  X   | X  |
Vollkorn                  |  4  |  X  |    |    |    |    |  X    |  X   |    |    |      |  X   | X  |
Webly&nbsp;Sleek          |  6  |     |    | X  |    |    |  X    |  X   |    | X  |      |  X   | X  |
Yanone&nbsp;Kaffeesatz    |  4  |     |    | X  |    |    |  X    |      |    | X  |  X   |  X   | X  |

#### Special Purpose Fonts

The fonts in this group are special purpose ones, i.e., they are usually
used only for particular headlines, subtitles, tags, etc. They are
usually just stand-alone fonts and cover just a very limited set of
Unicode characters.

Name                      |Fonts|Serif|Sans|Mono|Hand|Regular|Italic|Cond|Thin|Medium|Normal|Bold|Black
--------------------------|----:|:---:|:--:|:--:|:--:|:-----:|:----:|:--:|:--:|:----:|:----:|:--:|:---:
Amatic&nbsp;SC                 |  2  |     |    |    | X  |  X    |      |    |    |      |  X   |    |
Bebas&nbsp;Neue           |  1  |     | X  |    |    |  X    |      |    |    |      |      |    | X
Bellota                   |  6  |     |    |    |    |  X    |  X   |    |    |  X   |      | X  |
Bree&nbsp;Serif           |  1  |  X  |    |    |    |       |  X   |    |    |      |      | X  |
Chawp                     |  1  |     |    |    | X  |       |      |    |    |      |      | X  |
Comic&nbsp;Neue           | 12  |     | X  |    | X  |  X    |  X   |    | X  |      |  X   | X  |
Comic&nbsp;Relief         |  2  |     |    |    | X  |       |      |    |    |      |  X   | X  |
Courgette                 |  1  |     |    |    | X  |       |  X   |    |    |      |  X   |    |
Damion                    |  1  |     |    |    | X  |       |      |    |    |      |  X   |    |
DancingScript             |  2  |     |    |    | X  |       |      |    |    |      |  X   | X  |
Delius                    |  4  |     |    |    | X  |  X    |      |    |    |      |  X   | X  |
Grand&nbsp;Hotel          |  1  |     |    |    | X  |  X    |      |    |    |      |  X   |    |
Graziano                  |  1  |     |    |    | X  |       |  X   |    |    |      |  X   |    |
Great&nbsp;Vibes          |  1  |     |    |    | X  |       |  X   |    |    |      |  X   |    |
HH&nbsp;Samuel            |  1  |     | X  |    |    |  X    |      |    |    |      |      |    | X
Indie&nbsp;Flower              |  1  |     |    |    | X  |  X    |      |    |    |      |  X   |    |
Journal                   |  1  |     |    |    | X  |       |  X   |    |    |      |  X   |    |
Junction                  |  3  |     | X  |    |    |  X    |      |    |    |  X   |  X   | X  |
Kalam                     |  3  |     |    |    | X  |  X    |      |    |    |  X   |  X   | X  |
Kaushan Script            |  1  |     |    |    | X  |  X    |      |    |    |      |  X   |    |
Kingthings&nbsp;Petrock   |  2  |     |    |    | X  |  X    |      |    | X  |      |  X   |    |
Komika&nbsp;Hand          |  4  |     | X  |    | X  |  X    |  X   |    |    |      |  X   | X  |
Komika&nbsp;Text          | 10  |     | X  |    |    |  X    |  X   |    | X  |      |  X   | X  |
Learning&nbsp;Curve       |  2  |     |    |    | X  |       |  X   |    | X  |      |  X   |    |
Life Savers               |  3  |     |    |    | X  |       |  X   |    |    |      |  X   | X  | X
Lobster&nbsp;Two          |  4  |     |    |    | X  |  X    |  X   |    |    |      |  X   | X  |
Milonga                   |  1  |     |    |    | X  |  X    |      |    |    |      |  X   |    |
Nautilus&nbsp;Pompilius   |  1  |     |    |    | X  |  X    |      |    |    |      |  X   |    |
Bukhari&nbsp;Script       |  1  |     |    |    | X  |  X    |      |    |    |      |  X   |    |
Ostrich&nbsp;Sans         |  9  |     | X  |    |    |  X    |      |    | X  |  X   |  X   | X  | X
Patua&nbsp;One            |  1  |     |    |    | X  |       |  X   |    |    |      |  X   |    |
Poetsen                   |  1  |     |    |    | X  |  X    |      |    |    |      |      |    | X
Pompiere                  |  1  |     |    |    | X  |  X    |      |    |    |      |  X   |    |
Quando                    |  1  |     | X  |    |    |  X    |      |    |    |      |  X   |    |
Sansita&nbsp;One          |  1  |     |    |    | X  |       |  X   |    |    |      |      | X  |
Satisfy                   |  1  |     |    |    | X  |       |      |    |    |      |  X   |    |
Shadows&nbsp;Into&nbsp;Light&nbsp;Two    |  1  |     |    |    | X  |  X    |      |    |    |      |  X   |    |
Slabo                     |  2  |  X  |    |    |    |  X    |      |    |    |      |  X   |    |
To&nbsp;Be&nbsp;Continued |  1  |     |    |    | X  |  X    |      |    |    |      |      |    | X
Top&nbsp;Secret           |  1  |     | X  |    |    |  X    |      |    |    |      |      |    | X
Veteran&nbsp;Typewriter   |  1  |     |    | X  |    |       |      |    |    |      |  X   |    |
Yellowtail                |  1  |     |    |    | X  |       |  X   |    |    |      |  X   |    |

Browser Support
---------------

All major browsers are fully supported by TypoPRO due to the fact
that each font is provided in multiple formats. The actual minimum
browser version which supports a particular format follows (for latest
information see [1](http://caniuse.com/eot) [2](http://caniuse.com/ttf)
[3](http://caniuse.com/woff)):

Browser                     | EOT  | TTF  | WOFF
----------------------------|-----:|-----:|-----:
Internet Explorer (Desktop) | 6.0  |  9.0 |  9.0
Internet Explorer (Mobile)  | --   |  --  | 10.0
Mozilla Firefox (Desktop)   | --   |  3.5 |  3.6
Mozilla Firefox (Mobile)    | --   | 26.0 | 26.0
Google Chrome (Desktop)     | --   |  4.0 |  5.0
Google Chrome (Mobile 2)    | --   | 33.0 | 33.0
Google Chrome (Mobile 1)    | --   |  2.2 |  4.4
Apple Safari (Desktop)      | --   |  3.1 |  5.1
Apple Safari (Mobile)       | --   |  4.2 |  5.0
Opera (Desktop)             | --   | 10.0 | 11.1
Opera (Mobile)              | --   | 10.0 | 11.0

This means that for the latest versions of all major
browsers the optimized WOFF format would be both sufficient and
preferred. But for backward compatibility to older browser versions, the
EOT and TTF formats are still provided, too.

Notice: The fonts are intentionally not (or more precise: no longer since
TypoPRO 2.0) provided in the additional browser-supported SVG font
format. The reason simply is that since a longer time this gives not
really any additional effective cross-browser compatibility and just
unnecessarily increases the (already rather large) TypoPRO distribution
size.

Alternatives
------------

Nowadays, most of the fonts can also be individually
downloaded for free from _online_ font libraries like
[Font Squirrel](http://www.fontsquirrel.com/),
[DaFont](http://www.dafont.com/),
[Fonts2U](http://www.fonts2u.com) or
[fontsc](http://www.fontsc.com) or even _on-the-fly_ used from font
Cloud services like [Google Fonts](http://www.google.com/fonts),
[Open Font Library](http://openfontlibrary.org/),
[Adobe Edge Web Fonts](https://edgewebfonts.adobe.com/),
[Adobe Typekit](https://typekit.com/),
[FontDeck](http://fontdeck.com/) or
[Brick](http://brick.im/). But the purpose of **TypoPRO** is just
different: it provides an _offline_ solution and it intentionally
is a very opinionated pre-selection of reasonable (meaning
high-quality) fonts.

Download
--------

You can conveniently download snapshots or particular versions of **TypoPRO** in various ways:

- Bower: install as client component via the Bower component manager:<br/>
  `$ bower install typopro`<br/>
  `$ bower install "typopro#2.5.0"`

- NPM: install as server component via the Node Package Manager:<br/>
  `$ npm install typopro`<br/>
  `$ npm install typopro@2.5.0`

- cURL: download directly from the Github repository:<br/>
  `$ curl -O https://github.com/rse/typopro/archive/master.zip`<br/>
  `$ curl -O https://github.com/rse/typopro/archive/2.5.0.zip`

Alternatively, you can use [Grunt](http://gruntjs.com/) and
my companion [Grunt-TypoPRO](https://www.npmjs.org/package/grunt-typopro)
task for conveniently installing **TypoPRO** font families.

Desktop Usage
-------------

The DTP variants of all fonts you can find under `dtp/`. Just
install those files into your system:

- For Windows: For single and bulk installs, right-click one or more `*.ttf` files and choose "Install"
  from the context menu.

- For Mac OS X: For single installs, open the `*.ttf` file with the Font Book app
  and choose "Install Font". For bulk installs drag & drop the
  fonts into the `Fonts` folder of the `Library` folder in the
  Finder app.

- For GNU/Linux:
  For single installs, open the `*.ttf` file with the Font Viewer app
  and choose "Install". For bulk installs copy the
  `*.ttf` files into `/usr/local/share/fonts/truetype` folder of your system
  and run `sudo fc-cache`.

Web Usage
---------

It is important that the font files are delivered to the browser with
the correct MIME content-types. You can use the following Apache
`.htaccess` configuration snippet to achieve this:

    #   provide reasonable MIME content-types
    <IfModule mime_module>
        AddType application/vnd.ms-fontobject eot
        AddType application/font-woff         woff
        AddType application/x-font-ttf        ttf
    </IfModule>

    #   compress all fonts (except already compressed WOFF)
    <IfModule deflate_module>
        AddOutputFilter DEFLATE eot ttf
    </IfModule>

In order to use an individual font you have to use
two steps:

1. Include the `@font-face` based CSS font declaration:

    ```html
    <html>
        <head>
            [...]
            <link href="web/TypoPRO-Lora/TypoPRO-Lora-Regular.css" rel="stylesheet" type="text/css"/>
            [...]
        </head>
        [...]
    </html>
    ```

2. Use CSS to apply it to some of your HTML elements:

    ```html
    <html>
        <head>
            [...]
            <style type="text/css">
                .sample {
                    font-family: "TypoPRO Lora";
                    font-weight: bold;
                    font-size: 14pt;
                }
            </style>
            [...]
        </head>
        <body>
        [...]
            <div class="sample">The Quick Brown Fox Jumps Over The Lazy Dog</div>
        [...]
        </body>
    </html>
    ```

Recommendation
--------------

For general purpose typography, I can recommend you all of the above 60
General Purpose fonts, of course. But my personal preference most of
the times is towards the fonts in the following table. There are many

CHANGES.md

CHANGES
=======

2.5.0
-----

- added M+ font families

- added Inconsolata font

- added Andika New Basic font family

- added Bukhari Script

- added Roboto Mono font family

- updated Roboto font family to latest open source version from Google

- updated Aileron font family to latest version

- updated Charis SIL font family to latest version

2.4.1
-----

- added font Slabo 13px/27px

- added font Shadows Into Light Two

- added font Indie Flower

- added font Amatic SC

2.4.0
-----

- upgraded Fira Mono to version 3.200

- upgraded Fira Sans to version 4.100

- upgraded Roboto Sans to latest version

- upgraded Merriweather to latest version

- added fonts Montserrat and Montserrat Alteratives

- added font Domine

- added font Encode Sans

- added font Libre Caslon

- added font Life Savers

- added font Milonga

2.3.3
-----

- added Source Code Pro variant Office Code Pro

2.3.2
-----

- upgraded Lato to latest version
  (which has more fonts and dramatically more glyphs)

2.3.1
-----

- updated word cloud

- updated description

2.3.0
-----

- added Italic version of Source Code Pro

- added additional weights for Source Serif Pro

- added font Titillium

- added font PT Sans

- added font PT Serif

- added font PT Mono

2.2.8
-----

- added font family Quicksand (6 fonts)

2.2.6
-----

- added font family Asap (4 fonts)

2.2.5
-----

- added Cooper Hewitt font family

- added Bree Serif font

- updated Fira Sans and Fira Mono to latest version 3.111
  (includes a new font Fira Mono medium width)

2.2.4
-----

- added Quattrocento and Quattrocentro Sans font families

2.2.3
-----

- updated and cleaned up README

2.2.2
-----

- added font Bellota

- added font Damion

- added font FantasqueSansMono

- added font Satisfy

2.2.1
-----

- fixed packaging (the output files corresponding to all new fonts were
  not committed to Git and this way missing from the Bower deployment unit)

2.2.0
-----

- added font Kaushan Script

- added font Lekton

- added font Noto Sans

- added font Noto Serif

- added font Ek Mukta

- added font Kalam

2.1.2
-----

- updated Source Sans Pro to new version 2.010

- updated Roboto new version of 2014

2.1.1
-----

- added font face Rosario

- added font face Sansita One

2.1.0
-----

- rerendered all fonts with the latest fontface(1) utility
  which now sets the OS/2 FSType field to 0 to explicitly allow all
  fonts (in the DTP case) to be embedded into documents. For
  the WEB cases it does not matter, of course.

- added Signika font

2.0.6
-----

- updated Fira Sans from version 2.x to 3.105 and this way
  it now provides 16(!) font weights (we even had to workaround
  the fact that we have just 10 distinct weights in CSS).

- updated Fira Mono from version 2.x to 3.105.

- added Sinkin Sans font

2.0.5
-----

- add the brand-new Adobe Source Serif Pro font

2.0.4
-----

- disable auto-hinting on some fonts to avoid trouble with
  ttfautohint(1) version 1.1 and because it is unnecessary anyway.

- added the Poly font family.

- added the Liberation font family.

- added the Overlock font family.

- extended and polished some of blurb.txt files.

2.0.3
-----

- polish README

2.0.2
-----

- synchronized NPM and Bower distributions again.

2.0.1
-----

- republished with Node 0.10.28 / NPM 1.4.9 (instead of Node 0.11.13 /
  NPM 1.4.9) in order to get rid of the recent SHA1 checksum problems.

- fixed some file permissions to make NPM more happy

2.0.0
-----

- removed SVGZ format from Web fonts because SVGZ is really only
  needed for very ancient mobile devices and hence not worth the
  distribution extra space.

- removed HTML specimen files from Web fonts and replace them with a
  generic specimen.html at the top-level because this further reduces the
  redundancy and this way the distribution size.

- migrated from a generated index.html to a generic one which
  now loads a manifest.js (similar to the new specimen.html).

- fixed manifest: use the regular 'condensed' stretch value for 
  H.H. Samuel and not 'ultra-condensed'.

- provide a CSS snippet at the top right header in specimen.html
  for easy use of a particular TypoPRO font.

1.4.2
-----

- added fonts

1.4.1
-----

- ignore .git
- split Chrome for mobile
- add browser support information and polish some texts
- update cloud image
- add Ostrich Sans (Rounded) Heavy and Comic Relief
- add Alegreya, Comic Neue and Cuprum font families
- add more font directory references

1.4.0
-----

- add Webly Sleek, Ostrich Sans, Delius, To Be Continued and Comme fonts
- regenerate everything to benefit from latest fontface(1) improvements
- fix ordering
- split kerning and hinting descriptions
- improve markup
- better describe why hinting is required
- move format description into feature section
- add logo
- reduce the description to just one Manifest section
- fix description of directory layout
- provide hint how to download particular version
- provide hint how to download particular version
- improve documentation
- add cloud image

1.3.0
-----

- add Dosis, Junction, Lobster Two and Raleway
- fix Small-Caps variants by using an explicit name
- apply auto-hinting via ttfautohint

1.2.0
-----

- reconvert all files and reindex all fonts
- add links to the specimen
- better describe the distribution
- cleanup README text
- fix indentations
- add Aileron, Crimson, LatinModern, TeX Gyre, Nautilus Pompelius and EB Garamond

1.1.0
-----

- fix font ordering
- fix information about Droid
- add Dancing Script and Pompiere fonts
- add Libre Baskerville and Merriweather fonts
- provide some more font descriptions

1.0.0
-----

- add Aleo, Bitter, Courier Prime, Crete Round, Ostrich Sans and Oxygen fonts

0.9.8
-----

- regenerated all fonts with latest fontface tool

0.9.7
-----

- add Cabin, Courgette and Poetsen fonts

0.9.6
-----

- provide new DTP formats
- update Web formats and mix blurb/license files together
- rename dtp/ to src/ and generate a new special DTP set of fonts

etc/convert.sh
#!/bin/sh
##
##  TypoPRO - Fonts for Professional Typography
##  Copyright (c) 2013-2015 Ralf S. Engelschall <rse@engelschall.com>
##
##  Permission is hereby granted, free of charge, to any person obtaining
##  a copy of this software and associated documentation files (the
##  "Software"), to deal in the Software without restriction, including
##  without limitation the rights to use, copy, modify, merge, publish,
##  distribute, sublicense, and/or sell copies of the Software, and to
##  permit persons to whom the Software is furnished to do so, subject to
##  the following conditions:
##
##  The above copyright notice and this permission notice shall be included
##  in all copies or substantial portions of the Software.
##
##  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
##  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
##  MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
##  IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
##  CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
##  TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
##  SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
##

##
##  convert.sh: Convert all fonts into Web and DTP formats
##

#   U+0000..U+007F  Basic Latin                                128
#   U+0080..U+00FF  Latin-1 Supplement                         128
#   U+0100..U+017F  Latin Extended-A                           128
#   U+0180..U+024F  Latin Extended-B                           208
#   U+0250..U+02AF  IPA Extensions                              96
#   U+02B0..U+02FF  Spacing Modifier Letters                    80
#   U+0300..U+036F  Combining Diacritical Marks                112
#   --------------
#   U+1DC0..U+1DFF  Combining Diacritical Marks Supplement      64
#   U+1E00..U+1EFF  Latin Extended Additional                  256
#   --------------
#   U+2000..U+206F  General Punctuation                        112
#   U+2070..U+209F  Superscripts and Subscripts                 48
#   U+20A0..U+20CF  Currency Symbols                            48
#   U+20D0..U+20FF  Combining Diacritical Marks for Symbols     48
#   --------------
#   U+2150..U+218F  Number Forms                                64
#   --------------
#   U+2C60..U+2C7F  Latin Extended-C                            32
#   --------------
#   U+A720..U+A7FF  Latin Extended-D                           224

#   argument
pattern="$1"
if [ ".$pattern" = . ]; then
    pattern=".*"

    #   start with fresh output areas
    rm -rf web dtp
    mkdir web dtp
fi

#   configuration
prefix="TypoPRO"

#   sanity check execution
if [ ! -f etc/manifest.txt ]; then
    echo "ERROR: has to be called from top-level" 1>&2
    exit 1
fi

#   iterate over all fonts
OIFS="$IFS"; IFS="
"
for line in `cat etc/manifest.txt | egrep "$pattern"`; do
    IFS="$OIFS"

    #   parse font information
    font=`echo "$line" | sed -e 's; .*$;;'`
    family=`echo "$line" | sed -e 's;.*family="\([^"]*\)".*;\1;'`
    weight=`echo "$line" | sed -e 's;^;X;' -e 's;^X.*weight="\([^"]*\)".*;\1;' -e 's;^X.*;;'`
    style=`echo "$line" | sed -e 's;^;X;' -e 's;^X.*style="\([^"]*\)".*;\1;' -e 's;^X.*;;'`
    stretch=`echo "$line" | sed -e 's;^;X;' -e 's;^X.*stretch="\([^"]*\)".*;\1;' -e 's;^X.*;;'`
    variant=`echo "$line" | sed -e 's;^;X;' -e 's;^X.*variant="\([^"]*\)".*;\1;' -e 's;^X.*;;'`
    noautohint=`echo "$line" | sed -e 's;^;X;' -e 's;^X.*noautohint="\([^"]*\)".*;\1;' -e 's;^X.*;;'`
    name=`echo "$font" | sed -e 's;/.*;;'`
    echo "++ converting: $font ($family)"

    #   create font-specific target area
    if [ ! -d "web/$prefix-$name" ]; then
        mkdir web/$prefix-$name
    fi
    if [ ! -d "dtp/$prefix-$name" ]; then
        mkdir dtp/$prefix-$name
    fi

    #   determine font conversion command (common part)
    cmd="fontface"
    cmd="$cmd -l"
    if [ ".$noautohint" != .yes ]; then
        cmd="$cmd -h"
    fi
    cmd="$cmd -p \"$prefix\""
    cmd="$cmd -f \"$family\""
    if [ ".$weight" != . ]; then
        cmd="$cmd -W \"$weight\""
    fi
    if [ ".$style" != . ]; then
        cmd="$cmd -Y \"$style\""
    fi
    if [ ".$stretch" != . ]; then
        cmd="$cmd -S \"$stretch\""
    fi
    if [ ".$variant" != . ]; then
        cmd="$cmd -V \"$variant\""
    fi

    #   determine font conversion command (Web formats)
    cmd1="$cmd -u '0000-00FF'"
    cmd1="$cmd1 -O \"eot,ttf,woff,css\""
    cmd1="$cmd1 -o \"web/$prefix-$name/\""
    cmd1="$cmd1 \"src/$font\""

    #   determine font conversion command (DTP formats)
    cmd2="$cmd -u '0000-036F,1DC0-1EFF,2000-20FF,2150-218F,2C60-2C7F,A720-A7FF'"
    cmd2="$cmd2 -O \"ttf\""
    cmd2="$cmd2 -o \"dtp/$prefix-$name/\""
    cmd2="$cmd2 \"src/$font\""

    #   execute the font conversion commands
    eval $cmd1
    eval $cmd2
done
IFS="$OIFS"

#   aggregate the CSS files
for font in web/*; do
    name=`echo "$font" | sed -e 's;web/;;'`
    rm -f web/$name/$name.css
    cat web/$name/*.css >tmp.css
    mv tmp.css web/$name/$name.css
done

#   provide font information
for dir in src/*; do
    if [ ! -d $dir ]; then
        continue
    fi
    name=`echo "$dir" | sed -e 's;src/;;'`
    (   cat src/$name/blurb.txt
        echo ""
        cat src/$name/license.txt
    ) >web/$prefix-$name/$prefix-$name.txt
    (   cat src/$name/blurb.txt
        echo ""
        cat src/$name/license.txt
    ) >dtp/$prefix-$name/$prefix-$name.txt
done

